tests/sentry/snuba/test_discover.py

```diff
@@ -0,0 +1,390 @@
+            query="",
+            params={"project_id": [self.project.id]},
+        )
+        data = result["data"]
+        assert len(data) == 1
+        assert data[0]["project.id"] == self.project.id
+        assert data[0]["count_unique_user_email"] == 1
+
+    def test_field_aliasing_in_conditions(self):
+        result = discover.query(
+            selected_columns=["project.id", "user.email"],
+            query="user.email:bruce@example.com",
+            params={"project_id": [self.project.id]},
+        )
+        data = result["data"]
+        assert len(data) == 1
+        assert data[0]["project.id"] == self.project.id
+        assert data[0]["user.email"] == "bruce@example.com"
+
+    def test_release_condition(self):
+        result = discover.query(
+            selected_columns=["id", "message"],
+            query="release:{}".format(self.create_release(self.project).version),
+            params={"project_id": [self.project.id]},
+        )
+        assert len(result["data"]) == 0
+
+        result = discover.query(
+            selected_columns=["id", "message"],
+            query="release:{}".format(self.release.version),
+            params={"project_id": [self.project.id]},
+        )
+        assert len(result["data"]) == 1
+
+    def test_environment_condition(self):
+        result = discover.query(
+            selected_columns=["id", "message"],
+            query="environment:{}".format(self.create_environment(self.project).name),
+            params={"project_id": [self.project.id]},
+        )
+        assert len(result["data"]) == 0
+
+        result = discover.query(
+            selected_columns=["id", "message"],
+            query="environment:{}".format(self.environment.name),
+            params={"project_id": [self.project.id]},
+        )
+        assert len(result["data"]) == 1
+
+
+class QueryTransformTest(TestCase):
+    """
+    This test mocks snuba.raw_query to let us isolate column transformations.
+    """
+
+    @patch("sentry.snuba.discover.raw_query")
+    def test_query_parse_error(self, mock_query):
+        mock_query.return_value = {
+            "meta": [{"name": "transaction"}, {"name": "duration"}],
+            "data": [{"transaction": "api.do_things", "duration": 200}],
+        }
+        with pytest.raises(InvalidSearchQuery):
+            discover.query(
+                selected_columns=[],
+                query="foo(id):<1dino",
+                params={"project_id": [self.project.id]},
+            )
+        assert mock_query.call_count == 0
+
+    @patch("sentry.snuba.discover.raw_query")
+    def test_selected_columns_field_alias_macro(self, mock_query):
+        mock_query.return_value = {
+            "meta": [{"name": "user_id"}, {"name": "email"}],
+            "data": [{"user_id": "1", "email": "a@example.org"}],
+        }
+        discover.query(
+            selected_columns=["user", "project"], query="", params={"project_id": [self.project.id]}
```

Comment:
Do we want to assert somewhere that these get translated back to the correct public names rather than internal names?